ipns/ipns.go

```diff
@@ -0,0 +1,60 @@
+package ipns
+
+import (
+	"fmt"
+	"path"
+	"time"
+
+	shell "github.com/ipfs/go-ipfs-api"
+	"github.com/ipfs/go-ipns"
+	ipfscrypto "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/rs/zerolog/log"
+)
+
+type Manager interface {
+	UpdateIPNSEntry(string) error
+}
+
+type NopManager struct{}
+
+func NewNopManager() *NopManager {
+	return &NopManager{}
+}
+
+func (m *NopManager) UpdateIPNSEntry(commitCID string) error {
+	log.Debug().Msg("updating ipns entry in nop manager")
+	return nil
+}
+
+type IPNSManager struct {
```

Comment:
shouldn't it be an implementation of HeadManager?

Reply:
I think these two have distinct capabilities. We use HeadManager mainly to link commits and creating new ones, so they can be traversed. However IPNSManager makes an alias - so that head is, so to speak, constant.